### PR TITLE
Adding lat/lon values into node table

### LINKED ISSUE #4183: Develop latitude/longitude storage for tags
Develop latitude/longitude storage for tags.
 
>"The lon/lat storage i think would be useful for a few reasons. We do proximity right now based on truncating the lat/lon strings, which is a bit crude and can have some failures -- i think this is documented in an issue comment, actually. But let's discuss there! Mainly, we can't order by a string the way we could by a decimal column." @jywarren

* [x] Add fields to node and populate fields, retroactively, with the tag - lat/lon - values (#4292)
* [x] Store lat/lon into node - we should be looking in the node.rb add_tag method (#4379)
* [ ] Synchronize deletion between the two means of storing lat/lon - tag controller
* [ ] Prototype ways to subscribe to new node posts based on proximity
* [ ] Update/refactor `tagNearbyNodes()` API method that uses lat/lon node tags to use lat/lon node fields

### PR DESCRIPTION
**Pull request by milaaraujo**
Fixes #4183

### CODE REVIEW

**Comment by milaaraujo** [comment]
Hey @jywarren! When possible can you review this? I took the last days off, but I would like to get back to working on this as soon as possible!

**Comment by jywarren** [comment]
I hope to tomorrow! Coming back online then! Happy new year!!!

**Comment by jywarren** [comment]
OK, just a few CodeClimate changes to start with, should be pretty simple to resolve: https://codeclimate.com/github/publiclab/plots2/pull/4379

I'll do a close read in a bit!

File: app/models/node.rb [hunk 1]
```diff
@@ -750,6 +750,19 @@ def add_tag(tagname, user)
                                  uid: user.uid,
                                  date: DateTime.now.to_i,
                                  nid: id)
+
+          # Adding lat/lon values into node table
+          if tag.valid?
+            if tag.name.split(':')[0] == 'lat'
+              tagvalue = tag.name.split(':')[1]
+              self.update_attribute(:latitude, tagvalue)
```
Comment (jywarren) [inline app/models/node.rb]: OK this looks good but won't we need to change the type? to a decimal? 

Perhaps let's write a unit test to try this out - doing `node.add_tag('lat:10.0002', users(:bob))` and seeing if it adds the correct value and precision. If it fails due to the type not being correct, we can fix it after -- TDD!

File: app/models/node.rb [hunk 2]
```diff
@@ -750,6 +750,19 @@ def add_tag(tagname, user)
...
+              self.update_attribute(:precision, decimals(tagvalue).to_s)
+            elsif tag.name.split(':')[0] == 'lon'
+              tagvalue = tag.name.split(':')[1]
+              self.update_attribute(:longitude, tagvalue)
```
Comment (jywarren) [inline app/models/node.rb]: Here also we'll need to set precision.

File: app/models/node.rb [hunk 3]
```diff
@@ -750,6 +750,19 @@ def add_tag(tagname, user)
                                  uid: user.uid,
                                  date: DateTime.now.to_i,
                                  nid: id)
+
+          # Adding lat/lon values into node table
+          if tag.valid?
+            if tag.name.split(':')[0] == 'lat'
+              tagvalue = tag.name.split(':')[1]
+              self.update_attribute(:latitude, tagvalue)
+              self.update_attribute(:precision, decimals(tagvalue).to_s)
```
Comment (jywarren) [inline app/models/node.rb]: Ah, but is precision stored as a string? Shouldn't it be an int? Perhaps we should change it to an int; sorry if I missed this in the migration. Thanks!

**Comment by jywarren** [comment]
Thanks!!! Ok! That sounds great. I still think we should write a test to
assure the values are copied in, and then we can merge this. Thanks a lot!

**Comment by milaaraujo** [comment]
Not sure if we need a test if all we want to test is if we are not getting an error when adding the value. I'm testing here - creating a new note and adding the location - and the whole process seems to work just fine. I can attached here the screenshots of the process to show we are storing the correct values. Is it enough?

**Comment by jywarren** [comment]
I'd just like to have a unit test to ensure the latitude and longitude are
properly copied into the node record. This will help defend your code
against future code changes, too! In case someone in the future doesn't
understand this dual system. Thank you!!

**Comment by milaaraujo** [comment]
Ok, I can do it later then! 

Today I also want to start the PR to delete lat/lon. Where are we deleting the lat/lon tags? I didn’t find it.

**Comment by milaaraujo** [comment]
And now we have tests!

**Comment by jywarren** [comment]
Perfect!!!! Ok, merging this!

**Pull request merged**